src/pages/events/[id]/public-images.tsx

```diff
@@ -53,23 +102,61 @@ export default function Images() {
           <>
             <Header {...headerValues} />
             <div className="container">
-              <div className="w-auto p-4 text-2xl font-bold text-zinc-400">
-                <span>Images ({Images().length})</span>
+              <div className="w-auto p-4 text-2xl font-bold text-app-base-200">
+                <span>Images ({images.length})</span>
               </div>
               <div className="container mx-auto">
                 <div className="flex justify-end"></div>
               </div>
               <div className="grid gap-4 p-4 lg:grid-cols-5 ">
-                {Images()}
                 <div
-                  className="flex w-full items-center justify-center rounded-lg border bg-grey-100 p-4 shadow-md transition hover:-translate-y-1 hover:cursor-pointer hover:bg-secondary-50"
+                  className="flex w-full items-center justify-center rounded-lg border bg-app-base-100 p-4 shadow-md transition hover:-translate-y-1 hover:cursor-pointer hover:bg-app-secondary"
                   role="presentation"
+                  onClick={() => setShowModal(true)}
                 >
-                  <CiCirclePlus color="rgb(82 82 91)" size={100} />
+                  <CiCirclePlus className="text-app-base-200" size={100} />
                 </div>
+                {imagesToShow.map((image, index) => (
+                  <div key={index} className="w-40 h-30 flex items-center">
+                    <Image
+                      src={image}
+                      alt="Uploaded image"
+                      width={721}
+                      height={831}
+                      layout="intrinsic"
+                    />
+                  </div>
+                ))}
               </div>
-              <span className="w-full border-b-[1px] border-gray-200 p-2"></span>
+              <Modal
+                showModal={showModal}
+                setShowPopup={setShowModal}
+                title="Upload Images"
+              >
+                <form
+                  onSubmit={handleSubmit(onSubmit)}
+                  className="text-left p-4"
+                >
+                  <FieldImageUpload multiple name="images" control={control} />
+                  <div className="flex justify-center p-4">
+                    <div className="w-1/3">
+                      <button
+                        type="submit"
+                        className="bg-app-primary hover:bg-app-primary-focus text-white rounded h-9 w-full"
+                      >
+                        Submit
+                      </button>
+                    </div>
```

Comment:
The two parent divs for the button seems unnecessary. You could probably remove them. Maybe add `items-center` to the form and limit the button to `w-1/3`

Reply:
Moved `w-1/3` to button but didn't add `items-center` to the form as it also changes the styling of the image upload field.